packages/ui-components/src/components/Pagination/Pagination.test.tsx

```diff
@@ -432,10 +462,15 @@ describe("Pagination", () => {
     expect(screen.getByRole("textbox")).toHaveValue("")
   })
 
-  test("renders Pagination (select) with undefined totalPages prop as passed", () => {
-    act(() => {
-      render(<Pagination variant="select" totalPages={undefined} />)
-    })
+  test("renders Pagination (select) with undefined totalPages prop as passed", async () => {
+    await waitFor(() =>
```

Comment:
We usually use `waitFor` when we do assertions not when rendering a component. I'd suggest to keep `await act` and move `await waitFor` around `expect` statement below.

Reply:
IMO we don't even need a `waitFor`. Simply `await`ing for `act` would be sufficient, which wasn't the case before.

Reply:
@andypf true, but here I think we don't even need to use `waitFor`, wouldn't putting `await` in front of `act` work?